lib/internal/util.js

```diff
@@ -67,13 +68,16 @@ exports._deprecate = function(fn, msg) {
 };
 
 exports.decorateErrorStack = function decorateErrorStack(err) {
-  if (!(exports.isError(err) && err.stack))
+  if (!(exports.isError(err) && err.stack) ||
+      exports.getHiddenValue(err, 'decorated') === true)
     return;
 
   const arrow = exports.getHiddenValue(err, 'arrowMessage');
 
-  if (arrow)
+  if (arrow) {
     err.stack = arrow + err.stack;
+    exports.setHiddenValue(err, 'decorated', true);
```

Comment:
I'd probably give this a slightly more unique name, e.g. `'node:decorated'`.  (Ditto for `'arrowMessage'` although this PR didn't introduce that.)